src/components/common/Extensions/UserHub/partials/NotificationsTab/partials/Notification/Expenditure/ExpenditureNotification.tsx

```diff
@@ -0,0 +1,104 @@
+import React, { useMemo, type FC } from 'react';
+import { useNavigate } from 'react-router-dom';
+
+import {
+  type ColonyFragment,
+  useGetColonyActionQuery,
+  useGetExpenditureQuery,
+  useGetUserByAddressQuery,
+} from '~gql';
+import { TX_SEARCH_PARAM } from '~routes';
+import { type Notification as NotificationInterface } from '~types/notifications.ts';
+
+import NotificationWrapper from '../NotificationWrapper.tsx';
+
+import ExpenditureNotificationMessage from './ExpenditureNotificationMessage.tsx';
+
+const displayName =
+  'common.Extensions.UserHub.partials.ExpenditureNotification';
+
+interface NotificationProps {
+  colony: ColonyFragment | null | undefined;
+  loadingColony: boolean;
+  notification: NotificationInterface;
+}
+
+const ExpenditureNotification: FC<NotificationProps> = ({
+  colony,
+  loadingColony,
+  notification,
+}) => {
+  const navigate = useNavigate();
+
+  const { creator, expenditureID } = notification.customAttributes || {};
+
+  const { data: userData, loading: loadingUser } = useGetUserByAddressQuery({
+    variables: { address: creator || '' },
+    skip: !creator,
+  });
+  const { data: expenditureData, loading: loadingExpenditure } =
+    useGetExpenditureQuery({
+      variables: {
+        expenditureId: expenditureID || '',
+      },
+      skip: !expenditureID,
+    });
+
+  const creatorName = useMemo(() => {
+    return userData?.getUserByAddress?.items[0]?.profile?.displayName || '';
+  }, [userData]);
+
+  const expenditure = useMemo(() => {
+    return expenditureData?.getExpenditure;
+  }, [expenditureData]);
+
+  const transactionHash = useMemo(() => {
+    return expenditure?.creatingActions?.items[0]?.id;
+  }, [expenditure]);
```

Comment:
Are the useMemos necessary? expenditureData should already be memoized by useQuery hook

Reply:
No you're totally right, there're a leftover from when I was doing more with them!